frontend/src/ofrak/remote_resource.js

```diff
@@ -646,24 +646,26 @@ export class RemoteResource extends Resource {
   }
 
   async search_data(query, options) {
-    return await fetch(
-      `${this.uri}/search_data?type=${options.searchType}&regex=${options.regex}&caseignore=${options.caseIgnore}`,
-      {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify(query),
-      }
-    ).then(async (r) => {
+    return await fetch(`${this.uri}/search_data`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        search_query: query,
+        type: options.searchType,
+        regex: options.regex,
+        case_ignore: options.caseIgnore,
+      }),
```

Comment:
Unless there are other options that you don't want to pass, you can do the following. This will change the snake_case values the server is expecting to receive to camelCase ones, though. 

```suggestion
      body: JSON.stringify({
        search_query: query,
        ...options,
      }),
```

It has the effect of making this easier to change in the future – options don't need to be changed in this function anymore, only at the spots where they are passed in.

Reply:
love it